samples/kernel/condition_variables/pm/src/main.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <arch/cpu.h>
+#include <sys/arch_interface.h>
+#include <sys/atomic.h>
+
+#define NUM_THREADS 3
+#define TCOUNT 10
+#define COUNT_LIMIT 12
+
+static int state;
+enum power_state {
+	ACTIVE = 0,
+	SUSPENDING = 1,
+	SUSPENDED = 2,
+	RESUMING = 3,
+	OFF = 4
+};
+
+atomic_t usage_count;
+
+K_MUTEX_DEFINE(suspend_mutex);
+K_MUTEX_DEFINE(resume_mutex);
+
+K_CONDVAR_DEFINE(suspend_cv);
+K_CONDVAR_DEFINE(resume_cv);
+
+#define STACK_SIZE (1024 + CONFIG_TEST_EXTRA_STACKSIZE)
+
+K_THREAD_STACK_EXTERN(tstack);
+K_THREAD_STACK_ARRAY_DEFINE(tstacks, NUM_THREADS, STACK_SIZE);
+
+static struct k_thread t[NUM_THREADS];
+
+
+void resume(long thread_id)
+{
+	printk("Resuming by thread %ld..\n", thread_id);
+	state = RESUMING;
+	k_sleep(K_MSEC(500));
+	state = ACTIVE;
+	printk("Resumed by thread %ld...\n", thread_id);
+	k_condvar_broadcast(&resume_cv);
+}
+
+
+void suspend(long thread_id)
+{
+	printk("Suspending by thread %ld..\n", thread_id);
+	state = SUSPENDING;
```

Comment:
I believe this is wrong as it's modifying a variable that elsewhere is protected by a mutex.  There's also general advice to do the broadcast/signal while the mutex is held to ensure consistency in scheduling.  https://stackoverflow.com/questions/4544234/ has some relevant information.

Reply:
this sample was intended to show wentong how this could be used for PM and I forgot about it. I removed that sample now.